### ISSUE TITLE
Share alpha parameter across all conditions

### ISSUE THREAD

**Issue by jgallowa07** [matsengrp/multidms#217]
# Share alpha parameter across all conditions

## Summary

Replace the per-condition alpha parameters (`α_Delta`, `α_BA1`, `α_BA2`) with a single shared alpha value optimized jointly across all conditions. This eliminates the per-condition sigmoid degeneracy that causes alpha collapse at moderate-to-high fusionreg values.

## Background

PR #215 identified a persistent optimizer pathology: at fusionreg ≥ 1.6, the Delta condition's alpha collapses from ~6.0 to ~2.0 while the objective *decreases* — the optimizer is finding a genuinely lower-loss solution by exploiting the (alpha, beta0) degeneracy in the sigmoid. Stronger beta0_ridge (up to 100) and scale_fusion_by_n did not resolve this because:

- The sigmoid `g(φ) = expit(φ)` creates a manifold where `α * (g(φ) - g(φ_wt))` can produce the same predicted scores for different (α, β0) pairs
- Delta's wildtype operates at `sigmoid(φ_wt) ≈ 0.82` where the sigmoid gradient is weak (~0.15), making alpha collapse the path of least resistance when lasso drives shifts toward zero
- Per-condition alpha means each condition independently drifts on its own degenerate manifold

Sharing alpha across conditions anchors the parameter: BA.1 (the reference, with the most data) would anchor alpha, and Delta/BA.2 can no longer collapse independently. This is biologically reasonable since all conditions use the same DMS assay — the relationship between latent phenotype and functional score should be consistent.

## Proposed Approach

Change `Model.α` from `dict[str, Float[Array, ""]]` to a single `Float[Array, ""]`. The calibration block optimizes one alpha (plus per-condition logθ) instead of N independent alphas. All downstream code — prediction, trajectory logging, notebooks — is updated accordingly.

**Key design decisions:**
- **Not opt-in**: Change the model directly. If results are unsatisfactory, the branch simply won't merge.
- **Theta stays per-condition**: `logθ` is only relevant to count-based loss; per-condition overdispersion is physically meaningful.
- **Preserve existing results locally**: Before pulling new results, rename the local `results/` to `results_main/`. The pipeline `output_dir` stays as `results/` — no config change needed.

### 1. `jaxmodels.Model` — shared alpha field

**File:** `multidms/jaxmodels.py` (existing)

Change the `α` field from a dict to a scalar:

```python
class Model(eqx.Module):
    φ: dict[str, Latent]
    α: Float[Array, ""]  # was: dict[str, Float[Array, ""]]
    logθ: dict[str, Float[Array, ""]]
    ...
```

Update `predict_score` to use the single `self.α` for all conditions:

```python
def predict_score(self, data_sets):
    result = {}
    for d in data_sets:
        φ = self.φ[d]
        result[d] = self.α * (
            self.global_epistasis(φ(data_sets[d].X))
            - self.global_epistasis(φ(data_sets[d].x_wt))
        )
    return result
```

### 2. `jaxmodels.fit()` — initialization and calibration filter spec

**File:** `multidms/jaxmodels.py` (existing)

- `alpha_init` parameter changes from `dict[str, Float] | None` to `Float | None` (single scalar)
- Initialize: `α_val = jnp.array(alpha_init) if alpha_init is not None else jnp.array(1.0)`
- Model construction: `α=α_val` instead of `α=alpha_models`
- `filter_spec_calibration`: `α=True` (unchanged — equinox interprets `True` as marking all leaves trainable, which works for both dict and scalar pytree structures)

### 3. `jaxmodels.fit()` — trajectory logging

**File:** `multidms/jaxmodels.py` (existing)

Replace per-condition `alpha_{d}` columns with a single `alpha` column:

```python
# was: per_condition[f"alpha_{d}"] = float(model.α[d])
per_condition["alpha"] = float(model.α)
```

Update the column ordering list accordingly.

### 4. `model.py` — wrapper API

**File:** `multidms/model.py` (existing)

- `Model.__init__`: `alpha_init` parameter changes from dict to scalar
- Pass-through to `jaxmodels.fit()` updated accordingly
- Specific methods that access `self._jax_model.α[condition]` and must be updated to `self._jax_model.α`:
  - `get_variants_df()` (~line 422)
  - `predict()` (~line 712)

### 5. `model_collection.py` — fitting wrapper

**File:** `multidms/model_collection.py` (existing)

- `fit_one_model()`: `alpha_init` handling updated for scalar

### 6. Pipeline configs — separate output directory

**Files:**
- `experiments/scv2-spike/config/config.yaml`
- `experiments/scv2-spike/config/config_test.yaml`

- `output_dir` unchanged (`results` / `results_test`) — existing local `results/` renamed to `results_main/` before pulling new results
- `alpha_init: 6.0` (scalar instead of dict)

### 7. Pipeline notebooks — handle single alpha column

**Files:** `experiments/scv2-spike/notebooks/*.ipynb` and `_common.py`

- `_common.py`: `build_fit_params()` updates `alpha_init` to emit scalar
- `evaluate.ipynb` cell-7 (convergence diagnostics): replace `jm.alpha[cond]` dict access with scalar `jm.alpha`
- Evaluation/plotting notebooks: read `alpha` column instead of `alpha_{condition}` columns
- Alpha trajectory plots: single line instead of per-condition lines

## Testing Strategy

- **Existing tests**: All existing tests must pass after updating for the new alpha structure
- **Tests requiring update**:
  - `test_model_creation` (line 403): replace `assert len(model.α) == len(multi_condition_data)` with `assert model.α.shape == ()` since α is now a scalar, not a dict
- **New assertions**:
  - **Alpha type**: Verify `model.α` is a scalar `jnp.ndarray` with shape `()`
  - **Prediction equivalence**: When all per-condition alphas would have been identical, shared alpha should produce identical predictions
  - **Trajectory schema**: Verify trajectory DataFrame has `alpha` column (not `alpha_{condition}`)
- Edge cases:
  - Single-condition model (trivially shared)
  - `alpha_init=None` defaults to 1.0

## Documentation Updates

- [ ] Docstrings on `Model`, `fit()`, `alpha_init` parameter
- [ ] `model.py` wrapper docstrings
- [ ] Pipeline notebook markdown cells describing alpha

## Dependencies

No new dependencies.

### Alternative 1: Alpha ridge penalty

Add a penalty `λ_α * Σ (α_d - α_ref)²` analogous to `beta0_ridge`.

**Why not chosen:** Still allows per-condition drift — just penalizes it. Doesn't address the root degeneracy. Also adds another hyperparameter.

### Alternative 2: Hard alpha bounds / clamping

Clamp α ∈ [3, 12] during optimization.

**Why not chosen:** Arbitrary bounds that mask the underlying problem. Alpha would sit at the boundary, distorting gradients.

### Alternative 3: Opt-in flag (`share_alpha: bool`)

Make sharing alpha a configuration option defaulting to the old per-condition behavior.

**Why not chosen:** Adds complexity. This is an experimental branch — if it doesn't work, we don't merge. If it does work, per-condition alpha becomes dead code.

## Success Criteria

- [ ] All existing unit tests pass with shared alpha
- [ ] `pixi run spike-test` completes successfully
- [ ] Production pipeline (`results_shared_alpha/`) completes on remote server
- [ ] Alpha remains stable across the fusionreg grid (no collapse below 3.0)
- [ ] Stop mutation sparsity ≥ 95% at fusionreg ≥ 1.6
- [ ] Delta replicate shift correlations non-decreasing with fusionreg (tolerance of 0.02 for numerical noise)
- [ ] Existing local results preserved as `results_main/` before pulling new results

## Future Work

- If shared alpha resolves the collapse, consider FlexibleSigmoid (#197) as a further refinement
- Investigate whether shared alpha changes the optimal fusionreg selected by cross-validation
- Compare shift estimates at the CV-selected fusionreg between shared and per-condition alpha

## Enhancements to CLAUDE.md

- Update Architecture section: note that `Model.α` is a shared scalar, not per-condition
- Update "Key design patterns" to mention shared alpha as the default for Sigmoid GE
